syntax = "proto3";

package calculator;

// Định nghĩa dịch vụ Calculator
service Calculator {
    rpc Add (AddRequest) returns (AddResponse);
    rpc Subtract (SubtractRequest) returns (SubtractResponse);
    rpc Multiply (MultiplyRequest) returns (MultiplyResponse);
    rpc Divide (DivideRequest) returns (DivideResponse);
    rpc StreamAdd (stream AddRequest) returns (AddResponse);
}

message AddRequest {
    int32 a = 1;
    int32 b = 2;
}

message AddResponse {
    int32 result = 1;
}

message SubtractRequest {
    int32 a = 1;
    int32 b = 2;
}

message SubtractResponse {
    int32 result = 1;
}

message MultiplyRequest {
    int32 a = 1;
    int32 b = 2;
}

message MultiplyResponse {
    int32 result = 1;
}

message DivideRequest {
    int32 a = 1;
    int32 b = 2;
}

message DivideResponse {
    double result = 1;
}
